File: qc-core/src/main/java/com/eager/questioncloud/core/domain/userquestion/UserQuestionRepository.java
package com.eager.questioncloud.core.domain.userquestion;

import com.eager.questioncloud.core.domain.question.QuestionFilter;
import java.util.List;

public interface UserQuestionRepository {
    List<UserQuestion> saveAll(List<UserQuestion> userQuestionLibraries);

    Boolean isOwned(Long userId, List<Long> questionIds);

    Boolean isOwned(Long userId, Long questionId);

    List<UserQuestionDetail> getUserQuestions(QuestionFilter questionFilter);

    int countUserQuestions(QuestionFilter questionFilter);
}

File: qc-core/qc-question/qc-question-core/src/main/kotlin/com/eager/questioncloud/question/repository/UserQuestionRepository.kt
package com.eager.questioncloud.question.repository

import com.eager.questioncloud.common.pagination.PagingInformation
import com.eager.questioncloud.question.common.QuestionFilter
import com.eager.questioncloud.question.domain.UserQuestion
import com.eager.questioncloud.question.dto.UserQuestionContent

interface UserQuestionRepository {
    fun saveAll(userQuestionLibraries: List<UserQuestion>): List<UserQuestion>
    
    fun isOwned(userId: Long, questionIds: List<Long>): Boolean
    
    fun isOwned(userId: Long, questionId: Long): Boolean
    
    fun getUserQuestions(
        userId: Long,
        questionFilter: QuestionFilter,
        pagingInformation: PagingInformation
    ): List<UserQuestionContent>
    
    fun countUserQuestions(userId: Long, questionFilter: QuestionFilter): Int
    
    fun deleteAllInBatch()
    
    fun findByQuestionIdInAndUserId(questionIds: List<Long>, userId: Long): List<UserQuestion>
}
